src/server/graphql/mutations/downgradeToPersonal.ts

```diff
@@ -0,0 +1,78 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql'
+import stripe from 'server/billing/stripe'
+import getRethink from 'server/database/rethinkDriver'
+import DowngradeToPersonalPayload from 'server/graphql/types/DowngradeToPersonalPayload'
+import {sendAlreadyPersonalTierError} from 'server/utils/alreadyMutatedErrors'
+import {getUserId, getUserOrgDoc, isOrgBillingLeader} from 'server/utils/authorization'
+import {sendOrgLeadAccessError} from 'server/utils/authorizationErrors'
+import publish from 'server/utils/publish'
+import sendSegmentEvent, {sendSegmentIdentify} from 'server/utils/sendSegmentEvent'
+import {ORGANIZATION, PERSONAL, TEAM} from 'universal/utils/constants'
+
+export default {
+  type: DowngradeToPersonalPayload,
+  description: 'Downgrade a paid account to the personal service',
+  args: {
+    orgId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'the org requesting the upgrade'
+    }
+  },
+  async resolve (_source, {orgId}, {authToken, dataLoader, socketId: mutatorId}) {
+    const r = getRethink()
+    const now = new Date()
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+
+    // AUTH
+    const userId = getUserId(authToken)
+    const userOrgDoc = await getUserOrgDoc(userId, orgId)
+    if (!isOrgBillingLeader(userOrgDoc)) return sendOrgLeadAccessError(authToken, userOrgDoc)
+
+    // VALIDATION
+    const {stripeSubscriptionId, tier} = await r.table('Organization').get(orgId)
+
+    if (tier === PERSONAL) return sendAlreadyPersonalTierError(authToken, orgId)
+
+    // RESOLUTION
+    // if they downgrade & are re-upgrading, they'll already have a stripeId
+    const subscription = await stripe.subscriptions.del(stripeSubscriptionId)
```

Comment:
I imagine this is the spot that might need some graceful error handling if I've already nuked the sub

Reply:
Could you add an exception catch to this now?